### PR TITLE
Update links broken by migration

### PR DESCRIPTION
**Pull request by smelc**
# Changelog

```yaml
- description: |
    Make it possible to merge again, by fixing dead links
# uncomment types applicable to the change:
  type:
  # - feature        # introduces a new feature
  # - breaking       # the API has changed in a breaking way
  - compatible     # the API has changed but is non-breaking
  # - optimisation   # measurable performance improvements
  - improvement    # QoL changes e.g. refactoring
  # - bugfix         # fixes a defect
  # - test           # fixes/modifies tests
  # - maintenance    # not directly related to the code
  # - release        # related to a new release preparation
  # - documentation  # change in code docs, haddocks...
```

# Context

A number of links have been made 404 by the GitHub migration, and as the CI checks links for validity, this is blocking merging.

# How to trust this PR

Highlight important bits of the PR that will make the review faster. If there are commands the reviewer can run to observe the new behavior, describe them.

# Checklist

- [X] Commit sequence broadly makes sense and commits have useful messages
- NA New tests are added if needed and existing tests are updated. See [Running tests](https://github.com/input-output-hk/cardano-node-wiki/wiki/Running-tests) for more details
- [X] Self-reviewed the diff

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -30,4 +30,4 @@ See the [Contributing guide](CONTRIBUTING.md) for how to contribute to this proj
 
 Development documentation can be found in [Cardano Node Wiki](https://github.com/input-output-hk/cardano-node-wiki/wiki).
 
-Haddock documentation is available at: https://cardano-cli.cardano.intersectmbo.org/
+Haddock documentation is available at: https://intersectmbo.github.io/cardano-cli/
```
Comment (carbolymer) [inline README.md]: The correct url now is: https://cardano-api.cardano.intersectmbo.org/

**Review by carbolymer — CHANGES_REQUESTED** [review]
Afaik we should use domains within cardano.intersectmbo.org for api docs.

File: README.md (hunk 1 above)
Comment (smelc) [reply README.md]: Ah I see, this link was broken initially, but now it works. So this removes a change from this PR.

**Comment by smelc** [comment]
@carbolymer> ready for rereview, only one link change in the end.

**Review by carbolymer — APPROVED** [review]

**Comment by carbolymer** [comment]
Thanks @smelc!

**Pull request merged**